tests/common/include/tests/common/read_geometry.hpp

```diff
@@ -170,6 +177,69 @@ struct object_registry {
     }
 };
 
+// Minimalistic geometry type for toy geometry.
+template <typename volume_t, typename object_t,
+          template <typename...> class vector_type = dvector>
+struct toy_geometry {
+    // typedefs
+    using volume_type = volume_t;
+    using portal = object_t;
+    using portal_container = dvector<portal>;
+    using portal_links = typename object_t::edge_links;
+
+    struct object_registry {
+        using id = typename volume_type::objects;
+
+        template <typename value_type = void>
+        static constexpr auto get() {
+            return id::e_surface;
+        }
+    };
+
+    toy_geometry(vector_type<volume_t>&& volumes,
+                 vector_type<object_t>&& objects)
+        : _volumes(volumes), _objects(objects) {}
+
+    // data containers
+    vector_type<volume_t> _volumes;
+    vector_type<object_t> _objects;
+};
+
+// Minimalistic detector type for toy geometry
+template <typename geometry_t, typename transform_container_t,
```

Comment:
I guess the templeate parameter should be more expressive with `vector_type`

Reply:
Done